client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -306,7 +326,9 @@ export const siteSetupFlow: Flow = {
 					return navigate( 'options' );
 
 				case 'intent':
-					return navigate( isEnabled( 'signup/site-vertical-step' ) ? 'vertical' : 'intent' );
+					return navigate(
+						isEnabled( 'signup/site-vertical-step' ) ? 'vertical' : intentCaptureStep
+					);
```

Comment:
Given that this context is under `goBack()`, something feels off here.
1. When the current step is `intent`, we know that this is a version of the onboarding flow that doesn't have Goals Capture, and therefore replacing `intent` with `intentCaptureStep` is likely incorrect, as `intentCaptureStep` could be either `intent` or `goals`.
2. Goals Capture will be the first step, Verticals is the second step, so a case for handling go-back flow when `vertical` is the second step is likely needed.

```js
case 'intent':
	return navigate( isEnabled( 'signup/site-vertical-step' ) ? 'vertical' : 'intent' );
case 'vertical':
    // the latter part of is something I'm unsure of
    return navigate( isEnabled( 'signup/goals-step') ? 'goals' :  'vertical' ); 
```